### DISCUSSION TITLE
Allow next/jest to transform selected node_modules

### DISCUSSION THREAD

**Discussion by jimCresswell** [vercel/next.js#35363] [Ideas]
### Describe the feature you'd like to request

The next/jest config prevents any node_modules from being added to the transform list https://github.com/vercel/next.js/blob/acbd54322f447caaee09b1262e4687434ea0268c/packages/next/build/jest/jest.ts#L118

Some node_modules really do need transforming, e.g. https://github.com/sindresorhus/serialize-error/blob/main/index.js . This is a common enough pattern that it is explicitly mentioned in the Jest config docs https://jestjs.io/docs/configuration#transformignorepatterns-arraystring

Please allow the custom Jest config to override the next/jest assumptions

### Describe the solution you'd like

don't ignore custom transformIgnorePatterns like

```js
const customJestConfig = {
  transformIgnorePatterns: ["/node_modules/(?!(serialize-error)/)"],
};
```
it's a slightly tricky one, it might require a regex to detect node_modules matching patterns, or a separate config option in the `nextJest({...options})` invocation, e.g. `{transformNodeModules:["serialize-error"],}`.

### Describe alternatives you've considered

Setting up Jest manually

**Comment by icyJoseph** [comment]
I have observer this with a test we have for middleware, which imports from `next/server`. The current `next/jest` status, wraps the module with CommonJS (__dirname and the rest), and then it can't recognise the keyword `export` because it is an ES module.

I thought I could just make it ignore the patterns, but as far as I understand, there's an upper override on it?

Anyway, good to get this discussion started, I've seen other discussions opened with the same problem.

- https://github.com/vercel/next.js/discussions/34589
- https://github.com/vercel/next.js/discussions/34774 (has an actual work around!!! 🎉 )
- https://github.com/vercel/next.js/discussions/34589
